flow/compositor_context.h

```diff
@@ -48,6 +48,17 @@ enum class RasterStatus {
   kFailed
 };
 
+struct FrameDamage {
+  // IN: Previous layer tree
+  const LayerTree* prev_layer_tree = nullptr;
```

Comment:
This is a conceptual leftover from my previous attempt. Would `damage_rect` in your case mean `damage_out`? You still need to get one rect in, which is the accumulated damage for current framebuffer (in case of double / triple buffering)

Reply:
Isn't `Rasterizer::last_layer_tree` the last tree that was processed by the rasterizer? In the general case we may not be drawing to that same frame on the next pump. If there is more than one back buffer then we could be alternating and the tree we need to diff against would be the "second last" layer tree. I'm not sure if that happens on any of the platforms we work on in practice, but it is theoretically possible at least.

Also, it has been a while since I went through this code. I do know that there are frequently reasons to make us back out of rasterizing a frame and "try again" at which point we might decide to just jump to a new layer tree that has been delivered. How does `Rasterizer::last_layer_tree` track in those cases?

Reply:
@flar, double and triple buffering needs to be accounted for. I'm regularly testing this on Metal, which has probably the worst case of triple buffering, where sometimes one of the buffers gets really old. But you don't need to diff with anything else than last layer tree. You just need to accumulate additional damage for "inactive" framebuffers and then make sure you repaint that areas. The overhead is pretty small, especially for use-cases where it matter (small changes over frames).

Diffing with anything other than last LayerTree would be pretty difficult. The main reason why it is straightforward is that for every ContainerLayer framework reliably provides (with few fixes here and there) previous layer (the `oldLayer` argument in `SceneBuilder::push...` methods. If you want to diff two completely arbitrary LayerTrees, you don't get this anymore. Fortunately it's not necessary. 

My previous prototype could actually do very easy diff between arbitrary frames and I thought was a big advantage, but it wasn't. In fact, when I wanted to use this to invalidate backdrop snapshot, I realized that I actually need this/previous frame diff, regardless of what's in current framebuffer.

As for whether we can use `last_layer_tree`, that's TBD. If not, we can track it elsewhere. If we skip entire LayerTree during rasterization, the next one needs to be repainted fully. There's no saving that. However I don't think that should happen regularly, should it? Every time you skip rasterizing LayerTree would cause a jank.